atomic_reactor/inner.py

```diff
@@ -422,11 +422,24 @@ def build_docker_image(self):
         try:
             self.fs_watcher.start()
             signal.signal(signal.SIGTERM, self.throw_canceled_build_exception)
-            # time to run pre-build plugins, so they can access cloned repo
-            logger.info("running pre-build plugins")
             prebuild_runner = PreBuildPluginsRunner(self.builder.tasker, self,
                                                     self.prebuild_plugins_conf,
                                                     plugin_files=self.plugin_files)
+            buildstep_runner = BuildStepPluginsRunner(self.builder.tasker, self,
```

Comment:
Why was this changed?  What was wrong with creating each running and executing it?

Reply:
Currently, atomic-reactor loads plugins on a PluginsRunner initialization. It does so by loading all plugins available in the plugins path. It then makes all the plugins of the called runner type available to that runner.

This patch takes advantage of that work on initialization to also check if the requested plugins are available. Since this is done on a per runner basis, we now initialize all runners before running the first one.

Since we are not running each runner right after initializing it, I am also proposing to not reload all plugins every time we create a runner instance (current behavior). This will reduce the size of our logs.

In order to perform these changes, I had to make a few changes around the whole test suite, since most inner/plugin/plugins tests depend on the PluginsRunner initialization.